test/powershell/Modules/Microsoft.PowerShell.Management/Stop-Process.Tests.ps1

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Describe "Stop-Process" -Tags "CI" {
+    # Parameter validation
+    $command = Get-Command -Name Stop-Process
+
+    It "Should have an Id parameter" {
+        $command | Should -HaveParameter "Id" -Type [int[]] -Mandatory $false -DefaultValue $null
+    }
+
+    It "Should have an InputObject parameter" {
+        $command | Should -HaveParameter "InputObject" -Type [System.Diagnostics.Process[]] -Mandatory $false -DefaultValue $null
```

Comment:
We never do so formal tests.
Please add _functional_ tests.

Reply:
There should be a test where you create a process and a couple of child processes and verify that if the switch isn't used, the child processes don't get killed, but if the switch is used, the entire tree is killed.  Also, should have a test where a child process is not killed to verify the error handling.  Not sure if you can reliability create such a situation so you may need to have a test hook.  (search for test hooks in our current tests to see how they are used and how they are implemented)

Reply:
I guess the child process could be pwsh with Sleep 600.

Reply:
When writing the WIP PR I had similar thoughts, and pwsh also came to mind. I am however not quite as familiar with PowerShell on Mac and Linux. Would pwsh be the same on every environment?

Reply:
It looks like it is, I will try this.

Reply:
Windows and Unix-s differ in how they manage processes. You will most likely see differences in behavior and have differences in tests. For this reason, Steve mentioned hooks as a clue.